src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -907,6 +910,11 @@ private static bool IsDynamicStatics(TypeDesc type)
                     // OR 2. no code sharing lookup in call
                     // OR 3. we have have resolved to an instantiating stub
 
+                    // This check for introducing an instantiation stub comes from the logic at the very end of 
+                    // MethodTable::TryResolveConstraintMethodApprox under the conditional block
+                    // if (!pMD->GetMethodTable()->IsInterface()) around line 10050 in the file vm/methodtable.cpp.
```

Comment:
I'm not sure how much long-term value there is with comments like this. "This check is here because we copied it over from some other codebase at a specific point in time".

It might be better to say: `If we resolved the constrained call to a method on a value type, we need to provide the instantiation argument because it can't be acquired from "this"`.

Reply:
I have added "motivation comments" to both places based on your suggestion. I have kept the "correlation comments" though as I believe them to be of value. For Crossgen correlation, that is useful at least until Crossgen has been completely obsoleted and removed from any scenarios - I would be very surprised if that took less than half a year in a super-optimistic case. On top of that, neither comment actually points at Crossgen-specific codebase - they both belong to general CoreCLR runtime and I believe that we want to understand how we correlate with that.